invenio_rdm_records/ext.py

```diff
@@ -92,30 +93,74 @@ def init_config(self, app):
         ]
 
         for k in dir(config):
-            if k in supported_configurations or k.startswith('RDM_RECORDS_'):
+            if k in supported_configurations or k.startswith('RDM_'):
                 app.config.setdefault(k, getattr(config, k))
             if k in overriding_configurations and not app.config.get(k):
                 app.config[k] = getattr(config, k)
 
-    def _filter_record_service_config(self, app, service_config_cls):
-        """Filter record service config based on app global config."""
-        if not app.config["RDM_RECORDS_DOI_DATACITE_ENABLED"]:
-            service_config_cls.pids_providers.pop("doi", None)
-        return service_config_cls
+    def service_configs(self, app):
+        """Customized service configs."""
+        # Overall record permission policy
+        permission_policy = app.config.get("RDM_PERMISSION_POLICY", None)
+        doi_registration = app.config["RDM_RECORDS_DOI_DATACITE_ENABLED"]
+
+        # Available facets and sort options
+        sort_opts = app.config.get('RDM_RECORD_SORT_OPTIONS', None)
+        facet_opts = app.config.get('RDM_RECORD_FACETS', None)
+
+        # Specific configuration for each search endpoint
+        search_opts = SearchConfig(
+            app.config.get('RDM_RECORD_SEARCH'),
+            sort=sort_opts,
+            facets=facet_opts,
+        )
+        search_drafts_opts = SearchConfig(
+            app.config.get('RDM_RECORD_SEARCH_DRAFTS'),
+            sort=sort_opts,
+            facets=facet_opts,
+        )
+        search_versions_opts = SearchConfig(
+            app.config.get('RDM_RECORD_SEARCH_VERSIONING'),
+            sort=sort_opts,
+            facets=facet_opts,
+        )
+
+        class ServiceConfigs:
```

Comment:
could this become a de-facto `RDMServiceConfig`? Do we think this might be useful in another context?

Reply:
Hmm 🤔 Could be..... I must admit I mainly did it so I could have simple syntax in ``init_services()`` that didn't involve dicts..... went pretty fast.